thoth/python/source.py

```diff
@@ -189,7 +150,9 @@ def _warehouse_get_package_hashes(
             )
             # this checks whether to gather digests for all files in the given artifact
             if with_included_files:
-                result[-1]["digests"] = self._get_digests(item["url"], item["filename"])
+                artifact = PythonArtifact(item["filename"], item["url"], verify_ssl=self.verify_ssl)
```

Comment:
What about providing a `classmethod` which will instantiate `Artifact` from URL - `Artifact.from_url`? The `Source` class has already a method to construct URL (which is good to have at one place). The `Artifact` class would download it to a temporary location and keep path to it.

Reply:
I'm not sure what you mean?